GraphBase/src/stdio/ReaderWriter.java

```diff
@@ -0,0 +1,51 @@
+package stdio;
+import graph.Edge;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashSet;
+
+public class ReaderWriter {
+
+  public Collection<Edge> read(String fileName) throws IOException {
```

Comment:
You can read the graph data in and directly output Graph structure, either DirectedGraph or UndirectedGraph.